openmc/kinetics/solver.py

```diff
@@ -0,0 +1,831 @@
+from collections import OrderedDict
+import copy
+import os
+from scipy.sparse.linalg import spsolve
+from shutil import copyfile, move
+import subprocess
+import sys
+import time
+
+import h5py
+import numpy as np
+
+import openmc
+import openmc.kinetics
+from openmc.kinetics.clock import TIME_POINTS
+import openmc.mgxs
+
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class Solver(object):
+    """Solver to propagate the neutron flux and power forward in time.
+
+    Parameters
+    ----------
+    directory : str
+        A directory to save the transient simulation data.
+
+    Attributes
+    ----------
+    directory : str
+        A directory to save the transient simulation data.
+    shape_mesh : openmc.RegularMesh
+        Mesh by which shape is computed on.
+    unity_mesh : openmc.RegularMesh
+        Mesh with one cell convering the entire geometry.
+    amplitude_mesh : openmc.RegularMesh
+        Mesh by which amplitude is computed on.
+    tally_mesh : openmc.RegularMesh
+        Mesh by which to tally currents
+    geometry : openmc.geometry.Geometry
+        Geometry which describes the problem being solved.
+    settings_file : openmc.settings.SettingsFile
+        Settings file describing the general settings for each simulation.
+    materials_file : openmc.materials.MaterialsFile
+        Materials file containing the materials info for each simulation.
+    transient : OrderedDict()
+        Ordered dictionary describing the material changes during the transient.
+    mgxs_lib_file : openmc.materials.MGXSLibrary
+        MGXS Library file containing the multi-group xs for mg Monte Carlo.
+    clock : openmc.kinetics.Clock
+        Clock object.
+    one_group : openmc.mgxs.groups.EnergyGroups
+        EnergyGroups which specifies the a one-energy-group structure.
+    energy_groups : openmc.mgxs.groups.EnergyGroups
+        EnergyGroups which specifies the energy groups structure.
+    fine_groups : openmc.mgxs.groups.EnergyGroups
+        EnergyGroups used to tally the transport cross section that will be
+        condensed to get the diffusion coefficients in the coarse group
+        structure.
+    initial_power : float
+        The initial core power (in MWth).
```

Comment:
In the depletion module, we specify power in [W], so I would stick with the same convention here for consistency